src/tribler-core/tribler_core/modules/bandwidth_accounting/transaction.py

```diff
@@ -0,0 +1,200 @@
+
+        if self.sequence_number < 1:
+            return False
+
+        return True
+
+    def __eq__(self, other: BandwidthTransaction):
+        """
+        Check if a provided BandwidthTransaction is equal to this one.
+        :param other: The other bandwidth transaction.
+        """
+        return self.sequence_number == other.sequence_number and \
+               self.public_key_a == other.public_key_a and \
+               self.public_key_b == other.public_key_b and \
+               self.signature_a == other.signature_a and \
+               self.signature_b == other.signature_b and \
+               self.amount == other.amount and \
+               self.timestamp == other.timestamp
+
+    def __ne__(self, other: BandwidthTransaction):
+        """
+        Check if a provided BandwidthTransaction is not equal to this one.
+        :param other: The other bandwidth transaction.
+        """
+        return not self.__eq__(other)
+
+    @classmethod
+    def from_payload(cls, payload: BandwidthTransactionPayload) -> BandwidthTransaction:
+        """
+        Create a block according to a given payload.
+        This method can be used when receiving a block from the network.
+        :param payload: The payload to convert to a transaction.
+        :return A BandwidthTransaction, constructed from the provided payload.
+        """
+        return cls(payload.sequence_number, payload.public_key_a, payload.public_key_b,
+                   payload.signature_a, payload.signature_b, payload.amount, payload.timestamp)
+
+    @classmethod
+    def from_db(cls, db_obj: BandwidthTransaction) -> BandwidthTransaction:
+        """
+        Return a BandwidthTransaction object from a database object.
+        :param db_obj: The database object to convert.
+        :return A BandwidthTransaction object, based on the database object.
+        """
+        return BandwidthTransaction(db_obj.sequence_number, db_obj.public_key_a, db_obj.public_key_b,
+                                    db_obj.signature_a, db_obj.signature_b, db_obj.amount, db_obj.timestamp)
+
+    def get_db_kwargs(self) -> Dict:
+        """
+        Return the database arguments for easy insertion in a Pony database.
+        :return A dictionary with keyword arguments for database insertion.
+        """
+        return {
+            "sequence_number": self.sequence_number,
+            "public_key_a": self.public_key_a,
+            "public_key_b": self.public_key_b,
+            "signature_a": self.signature_a,
+            "signature_b": self.signature_b,
+            "amount": self.amount,
+            "timestamp": self.timestamp
+        }
+
+
+def define_binding(bandwidth_database):
+    db = bandwidth_database.database
+
+    class BandwidthTransaction(db.Entity):
+        """
+        This class describes a bandwidth transaction that resides in the database.
+        """
+        sequence_number = Required(int)
+        public_key_a = Required(bytes)
```

Comment:
Well, we **really** want indexes on public keys, sequence_number, timestamp, etc. Like on all of these fields, except for the signature. And just for safety, it is better to make signature fields `unique`

Reply:
@ichorid, unique indexes for signature fields can be detrimental for insert performance.
Also, what about a very small chance for two different transactions to have the same signature? Do we want to receive an exception while adding the second transaction to the database?

Reply:
Since our queries are only using the public keys currently, I have added indices on the public keys.